tests/poll-until.spec.ts

```diff
@@ -0,0 +1,44 @@
+import {pollUntil} from '../source/ts/poll';
+
+const wait = async (milliseconds: number) =>
+  new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+describe('using pollUntil to see if a function matches a value', () => {
+  const functionThatReturns42 = () => Promise.resolve(42);
+  const timeoutMs = 10;
+
+  it('returns true if the expected value is returned within the timeout time', async () => {
+    const didReturnExpectedValue = await pollUntil(
+      (currentResult) => currentResult === 42,
+      () => functionThatReturns42(),
```

Comment:
Could just be `functionThatReturns42`, as it's already a function

Reply:
f5247ab93c8798682c5adc136dd727bc6b29fb59